internal/store/elasticsearch/discovery_search_repository.go

```diff
@@ -249,22 +249,25 @@ func (repo *DiscoveryRepository) buildFunctionScoreQuery(query elastic.Query, ra
 		Weight(1.0)
 
 	fsQuery := elastic.NewFunctionScoreQuery().
-		ScoreMode(defaultFunctionScoreQueryScoreMode).
+		Add(
+			elastic.NewTermQuery("name.keyword", text),
+			elastic.NewWeightFactorFunction(2),
+		).
 		AddScoreFunc(factorFunc).
-		Query(query)
-
+		Query(query).
+		ScoreMode(defaultFunctionScoreQueryScoreMode)
```

Comment:
Don't fully understand the changes made here. Hope the changes have been tested.

Reply:
These changes have been tested. There were cases when exact match is not coming on top, hence we added term query. I have highlighted the changes in query in bold here.
"function_score":
        {
            "functions":
            [
                **{
                    "filter":
                    {
                        "term":
                        {
                            "name.keyword": "detail_gopay_consolidated_gtv"
                        }
                    },
                    "weight": 2
                },**
                {
                    "field_value_factor":
                    {
                        "field": "data.stats_metadata.query_count",
                        "missing": 1,
                        "modifier": "log1p"
                    },
                    "weight": 1
                }
            ],

Reply:
@sudo-suhas : We can connect if any further clarification is required.